### PR TITLE
Add interface for I18n's date time classes.

### PR DESCRIPTION
**Pull request by ADmad**
This helps in typehinting and also allows to better "hide" Chronos. Lot of users are confused regarding the various I18n classes and Chronos classes.

### CODE REVIEW

**Comment by ADmad** [comment]
The `Cookie` class uses `Chronos` to create datetime instances for `expiresAt`. Is using `Chronos` really necessary there? At some point I would like to have the Http client available as a standalone package and this usage would add unnecessary dependency on Chronos.

**Comment by saeideng** [comment]
> Is using Chronos really necessary there? 

you can replace it with `DateTime/DateTimeImmutable`

**Comment by markstory** [comment]
I like the idea. Having an interface does help reduce typehinting on concrete types.

File: src/I18n/TimeInterface.php [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+     */
+    public static function parseTime(string $time, $format = null);
+
+    /**
+     * Get the difference formatter instance or overwrite the current one.
+     *
+     * @param \Cake\I18n\RelativeTimeFormatter|null $formatter The formatter instance when setting.
+     * @return \Cake\I18n\RelativeTimeFormatter The formatter instance.
+     */
+    public static function diffFormatter($formatter = null);
```
Comment (dereuromark) [inline src/I18n/TimeInterface.php]: shouldnt this also be a split getter setter? and can we typehint the class?

**Comment by ADmad** [comment]
This should be good to go. I'll deal with `Cookie`'s use of `Chronos` in a separate PR.

**Pull request merged**